src/main.rs

```diff
@@ -32,6 +33,59 @@ type PackageName<'a> = &'a str;
 type RootFeature<'a> = (PackageName<'a>, Feature<'a>);
 
 fn main() {
+    let args: Vec<String> = std::env::args().collect();
+    let args: Vec<&str> = args.iter().map(AsRef::as_ref).collect();
+
+    match args.as_slice() {
+        [_, "--stdin"] => generate_cargo_nix(io::stdout()),
+        [_, "--file"] => write_to_file("Cargo.nix"),
+        [_, "--file", file] => write_to_file(file),
+        [_] => print_help(),
+        [_, "--help"] => print_help(),
+        [_, "--version"] => print_version(),
+        _ => {
+            println!("Invalid arguments: {:?}", &args[1..]);
+            println!("\nTry again, with help: \n");
+            print_help();
+        }
+    }
+}
+
+fn print_version() {
+    println!("cargo2nix-{}\n", env!("CARGO_PKG_VERSION"));
+}
+fn print_help() {
+    print_version();
+    println!("$ cargo2nix                     # Print the help");
```

Comment:
Oops it should have been `--stdout` instead of `--stdin`. If you want that to be removed, then why do you think printing the output to `stdout` is a good idea without `--stdout`?

Reply:
Changed `--stdin` to `--stdout` only... and resolving this issue until you disagree and wanna discuss it.